Update php docs regarding to the sdk imrpovements (#80)
configcat/docs: website/docs/sdk-reference/php.md

| Name      | Type     | Description                                                                                               |
| --------- | -------- | --------------------------------------------------------------------------------------------------------- |
| `sdkKey`  | `string` | **REQUIRED.** SDK Key to access your feature flag and setting. Get it from *ConfigCat Dashboard*. |
| `options` | `array`  | **Optional.** Additional SDK options, see below for the detailed list.                          |

Available options:

| Name                     | Type                           | Description                         |
| ------------------------ | ------------------------------ | ----------------------------------- |
| `data-governance`        | `int`                          | Optional, defaults to `DataGovernance::GLOBAL_`. Describes the location of your feature flag and setting data within the ConfigCat CDN. This parameter needs to be in sync with your Data Governance preferences. [More about Data Governance](advanced/data-governance.md). Available options: `GLOBAL_`, `EU_ONLY`. |

> | `logger`                 | `\Psr\Log\LoggerInterface`     | Optional, configures a logger for errors and warnings produced by the SDK, defaults to [Monolog](https://github.com/Seldaek/monolog). [See below](#logging). |

Reviewer: ```suggestion
| `logger`                 | `\Psr\Log\LoggerInterface`     | Optional, configures a logger for errors and warnings produced by the SDK, defaults to [Monolog](https://github.com/Seldaek/monolog). [More about logging](#logging). |
```

> | `log-level`              | `int`                          | Optional, defaults to `LogLevel::WARNING`. Sets the internal log level. [See below](#logging). |

Reviewer: ```suggestion
| `log-level`              | `int`                          | Optional, defaults to `LogLevel::WARNING`. Sets the internal log level. [More about log levels](#logging). |
```

> | `cache`                  | `\ConfigCat\Cache\ConfigCache` | Optional, sets a `\ConfigCat\Cache\ConfigCache` implementation for caching the latest feature flag and setting values. [See below](#cache). You can check the currently available implementations [here](https://github.com/configcat/php-sdk/tree/master/src/Cache). |

Reviewer: ```suggestion
| `cache`                  | `\ConfigCat\Cache\ConfigCache` | Optional, sets a `\ConfigCat\Cache\ConfigCache` implementation for caching the latest feature flag and setting values. [More about cache](#cache). You can check the currently available implementations [here](https://github.com/configcat/php-sdk/tree/master/src/Cache). |
```

| `cache-refresh-interval` | `int`                          | Optional, sets the refresh interval of the cache in seconds, after the initial cached value is set this value will be used to determine how much time must pass before initiating a [config.json download](/requests). Defaults to 60. |
| `request-options`        | `array`                        | Optional, sets the request options (e.g. [HTTP Timeout](#http-timeout), [HTTP Proxy](#http-proxy)) for the underlying `Guzzle` HTTP client used for [downloading the config.json](/requests) files. See Guzzle's [official documentation](https://docs.guzzlephp.org/en/stable/request-options.html) for the available request options. |

> | `flag-overrides`         | `\ConfigCat\Override\OverrideDataSource` | Optional, configures local feature flag & setting overrides. [See below](#flag-overrides). |

Reviewer: ```suggestion
| `flag-overrides`         | `\ConfigCat\Override\OverrideDataSource` | Optional, configures local feature flag & setting overrides. [More about feature flag overrides](#flag-overrides). |
```

| `exceptions-to-ignore`   | `array`                        | Optional, sets an array of exception classes that should be ignored from logs. |

> | `base-url`               | `string`                       | Optional, sets the CDN base url (forward proxy, dedicated subscription) from where the sdk will download the feature flag and setting. |

Reviewer: ```suggestion
| `base-url`               | `string`                       | Optional, sets the CDN base url (forward proxy, dedicated subscription) from where the SDK will download the features flag and settings. |
```

> Available configuration options:

Author: Done

| Name                     | Type                           | Description                         |
| ------------------------ | ------------------------------ | ----------------------------------- |
| `data-governance`        | `int`                          | Optional, defaults to `DataGovernance::GLOBAL_`. Describes the location of your feature flag and setting data within the ConfigCat CDN. This parameter needs to be in sync with your Data Governance preferences. [More about Data Governance](advanced/data-governance.md). Available options: `GLOBAL_`, `EU_ONLY`. |

> | `logger`                 | `\Psr\Log\LoggerInterface`     | Optional, configures a logger for errors and warnings produced by the SDK, defaults to `Psr\Log\NullLogger`. |

Author: That is not part of the SDK's source, it's the standard logging interface, but I added a link to the logging section

> | `logger`                 | `\Psr\Log\LoggerInterface`     | Optional, configures a logger for errors and warnings produced by the SDK, defaults to `Psr\Log\NullLogger`. |

Reviewer: `\ConfigCat\Cache\ConfigCache`
The ones similar to these can be linked.

> | `logger`                 | `\Psr\Log\LoggerInterface`     | Optional, configures a logger for errors and warnings produced by the SDK, defaults to `Psr\Log\NullLogger`. |

Author: Turned them into links, the 2 new ones will point to 404 as they are not on master yet

| `log-level`              | `int`                          | Optional, defaults to `LogLevel::WARNING`. Sets the internal log level. |
| `cache`                  | `\ConfigCat\Cache\ConfigCache` | Optional, sets a `\ConfigCat\Cache\ConfigCache` implementation for caching the actual configurations. You can check the currently available implementations [here](https://github.com/configcat/php-sdk/tree/master/src/Cache). |

> | `cache-refresh-interval` | `int`                          | Optional, sets the refresh interval of the cache in seconds, after the initial cached value is set this value will be used to determine how much time must pass before initiating a new configuration fetch request. Defaults to 60. |

Author: Done

> | `request-options`        | `array`                        | Optional, sets the options for the request initiated by the `Guzzle` HTTP client. See the [official documentation](https://docs.guzzlephp.org/en/stable/request-options.html) for the available options. |

Author: Rephrased, plus added more info

```

## `getAllKeys()`
You can query the keys of each feature flag and setting with the `getAllKeys()` method.

```php
$client = new \ConfigCat\ConfigCatClient("#YOUR-SDK-KEY#");
$keys = $client->getAllKeys();
```

## Flag Overrides

> With flag overrides you can overwrite the feature flags & settings fetched from the ConfigCat CDN with local values.

Reviewer: ```suggestion
With flag overrides you can overwrite the feature flags & settings downloaded from the ConfigCat CDN with local values.
```

Reviewer: Please replace everywhere

$keys = $client->getAllKeys();
```

## Flag Overrides

> With flag overrides you can overwrite the feature flag & setting configuration fetched from the ConfigCat CDN with local values.

Reviewer: ```suggestion
With flag overrides you can overwrite the feature flag & setting configuration downloaded from the ConfigCat CDN with local values. This could be useful when ... 
```

You can load your feature flag & setting overrides from a file or from a simple associative array.

### JSON File

The SDK can be configured to load your feature flag & setting overrides from a file or classpath resource. 
You can also specify whether the file should be reloaded when it gets modified.
```php
$client = new ConfigCatClient("localhost", [
  ClientOptions::FLAG_OVERRIDES => OverrideDataSource::localFile(
    "tests/test-rules.json", // path to the file
    OverrideBehaviour::LOCAL_ONLY // local/offline mode
  ),
]);
```
#### JSON File Structure
The SDK supports 2 types of JSON structures to describe feature flags & settings.

##### 1. Simple (key-value) structure
```json
{
  "flags": {
    "enabledFeature": true,
    "disabledFeature": false,
    "intSetting": 5,
    "doubleSetting": 3.14,
    "stringSetting": "test"
  }
}
```

##### 2. Complex (full-featured) structure
This is the same format that the SDK fetches from the ConfigCat CDN. 

> It allows the usage of all features you can do on the ConfigCat Dashboard.

Reviewer: The URL would be helpful from where they can download their current config.json file so they can re-write it.

Author: done

                        // 0  -> 'IS ONE OF',
                        // 1  -> 'IS NOT ONE OF',
                        // 2  -> 'CONTAINS',
                        // 3  -> 'DOES NOT CONTAIN',
                        // 4  -> 'IS ONE OF (SemVer)',
                        // 5  -> 'IS NOT ONE OF (SemVer)',
                        // 6  -> '< (SemVer)',
                        // 7  -> '<= (SemVer)',
                        // 8  -> '> (SemVer)',
                        // 9  -> '>= (SemVer)',
                        // 10 -> '= (Number)',
                        // 11 -> '<> (Number)',
                        // 12 -> '< (Number)',
                        // 13 -> '<= (Number)',
                        // 14 -> '> (Number)',
                        // 15 -> '>= (Number)',
                        // 16 -> 'IS ONE OF (Sensitive)',
                        // 17 -> 'IS NOT ONE OF (Sensitive)'
                    "c": "@example.com", // comparison value
                    "v": true, // value served when the rule is selected during evaluation
                    "i": "bcfb84a7" // variation id (for analytical purposes)
                }
            ]
        },
    }
}
```

> > For a baseline, you can download your configuration from ConfigCat's CDN and use it as-is in a file.

Reviewer: I suggest moving this line under the `complex` title

## Logging
The SDK uses the PSR-3 `LoggerInterface` for logging, so you can use any implementor package like [Monolog](https://github.com/Seldaek/monolog).
```php
$client = new \ConfigCat\ConfigCatClient("#YOUR-SDK-KEY#", [
    \ConfigCat\ClientOptions::LOGGER => new \Monolog\Logger("name"),
]);
```
You can change the verbosity of the logs by setting the `log-level` configuration option.
```php
$client = new \ConfigCat\ConfigCatClient("#YOUR-SDK-KEY#", [
    \ConfigCat\ClientOptions::LOG_LEVEL => \ConfigCat\Log\LogLevel::INFO
]);
```

The following levels are used by the ConfigCat SDK:

| Level      | Description                                                                             |
| ---------- | --------------------------------------------------------------------------------------- |
| `NO_LOG`   | Turn the logging off.                                                         |
| `ERROR`    | Only error level events are logged.                                                     |

> | `WARNING`  | Errors and Warnings are logged.                                                         |

Reviewer: ```suggestion
| `WARNING`  | Default. Errors and Warnings are logged.                                                         |
```

| `INFO`     | Errors, Warnings and feature flag evaluation is logged.                                 |
| `DEBUG`    | All of the above plus debug info is logged. Debug logs can be different for other SDKs. |

> > The default level is `WARNING`. With `NO_LOG` you can turn the logging completely off.

Reviewer: this line blends in with the rest of the text. I suggest removing or coloring.

Author: Done

> Info level logging helps to inspect the feature flag evaluation process:

Reviewer: ```suggestion
Info level logging helps to inspect how a feature flag was evaluated:
```

```bash
[2022-01-06T18:34:16.846039+00:00] ConfigCat.INFO: Evaluating getValue(isPOCFeatureEnabled).
User object: {"Identifier":"435170f4-8a8b-4b67-a723-505ac7cdea92","Email":"john@example.com"}
Evaluating rule: [Email:john@example.com] [CONTAINS] [@something.com] => no match.
Evaluating rule: [Email:john@example.com] [CONTAINS] [@example.com] => match, returning: true.
```

## HTTP Client

> The SDK uses [Guzzle](http://docs.guzzlephp.org/en/stable/index.html) for the underlying HTTP calls and its request options are available to customize through SDK options.

Reviewer: I suggest explaining:
Why would I want to do it?
How to do it?

Author: The next 2 sections below explain what you can do in which scenarios

### HTTP Proxy
If your application runs behind a proxy you can do the following:
```php
$client = new \ConfigCat\ConfigCatClient("#YOUR-SDK-KEY#", [
    \ConfigCat\ClientOptions::REQUEST_OPTIONS => [
        \GuzzleHttp\RequestOptions::PROXY => [
            'http'  => 'tcp://localhost:8125',
            'https' => 'tcp://localhost:9124',
        ]
    ],
]);
```
### HTTP Timeout

> You can set the maximum wait time for a ConfigCat HTTP response by using Guzzle's timeouts.

Reviewer: I suggest explaining:
What does maximum wait time mean?

Author: Feel free to suggest your version, I don't see the reason to explain a timeout in more detail.
